build/build.go

```diff
@@ -447,6 +447,25 @@ func toSolveOpt(ctx context.Context, node builder.Node, multiDriver bool, opt Op
 		so.FrontendAttrs["multi-platform"] = "true"
 	}
 
+	attests := make(map[string]string)
+	for k, v := range opt.Attests {
+		if v != nil {
+			attests[k] = *v
+		}
+	}
+	supportsAttestations := bopts.LLBCaps.Contains(apicaps.CapID("exporter.image.attestations"))
+	if len(attests) > 0 {
+		if !supportsAttestations {
+			return nil, nil, errors.Errorf("attestations are not supported by the current buildkitd")
+		}
+		for k, v := range attests {
+			so.FrontendAttrs[k] = v
+		}
+	}
+	if _, ok := opt.Attests["attest:provenance"]; !ok && supportsAttestations {
+		so.FrontendAttrs["attest:provenance"] = "mode=min,inline-only=true"
```

Comment:
I'm not sure I understand what stops the inline attestation to be added when loading to docker? Or did we block this in buildkit side?

Reply:
Yup, this is blocked buildkit side - https://github.com/moby/buildkit/pull/3466/files#diff-93b81ebb2d74a7ff89a643dfd137603e309b87754076de895b0c15dd4626a169R76.

We will attach inline attestations - but only if we would are already exporting an image index. For cases where we need to transform the manifest into the index, we only perform that upgrade for the `image` exporter.